### PR TITLE
PR-2211 Add llapi_path2fid support

### PR DESCRIPTION
**Pull request by beevans**
Signed-off-by: Ben Evans <beevans@whamcloud.com>

---
This change is [Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/2212)

### CODE REVIEW

File: liblustreapi/src/lib.rs [hunk 1]
```diff
@@ -266,10 +266,66 @@ impl Llapi {
...
+    }
+
+    pub fn search_fsname(&self, pathname: &str) -> Result<String, LiblustreError> {
+	let fsc = CString::new(pathname.as_bytes())?;
+	let mut page: Vec<u8> = vec![0; std::mem::size_of::<u8>() * 16];
+	let ptr = page.as_mut_ptr() as *mut libc::c_char;
+
+	let rc = unsafe {
+	    match self.lib.get(b"llapi_search_fsname\0") {
+		Ok(f) => {
```
Comment (utopiabound) [inline liblustreapi/src/lib.rs:308]: (same as above)

**Review by utopiabound — CHANGES_REQUESTED** [review]

**Comment by jgrund** [comment]
Looks like this code needs `rustfmt` run on it.

File: liblustreapi/src/lib.rs [hunk 2]
```diff
@@ -266,10 +266,66 @@ impl Llapi {
...
     }
 
-    pub fn search_rootpath(&self, fsname: &str) -> Result<String, LiblustreError> {
-        // @TODO this should do more validation
+    pub fn path2fid(&self, _mntpt: &str, path: &PathBuf) -> Result<String, LiblustreError> {
+	let pathptr = CString::new(path.to_str().unwrap())?.into_raw();
+	let mut fid = Fid{ seq: 0, oid: 0, ver: 0,};
+	
+        let rc = unsafe {
+            let rc = match self.lib.get(b"llapi_path2fid\0") {
```
[the remark this answers is not in the document]
Comment (jgrund) [reply liblustreapi/src/lib.rs:274]: I think we should add a `SymbolNotFound` variant to `LiblustreError` so it's clear to the caller what went wrong.

[the remark this answers is not in the document]
Comment (beevans) [reply liblustreapi/src/lib.rs:274]: I'm just following the pattern of fid2path

[the remark this answers is not in the document]
Comment (jgrund) [reply liblustreapi/src/lib.rs:278]: yeah, looks like that should be updated as well.

File: liblustreapi/src/lib.rs [hunk 3]
```diff
@@ -266,10 +266,66 @@ impl Llapi {
...
+
+        if rc != 0 {
+            return Err(LiblustreError::os_error(rc.abs()));
+        }
+	Ok(fid.to_string())
+    }
+
+    pub fn search_fsname(&self, pathname: &str) -> Result<String, LiblustreError> {
+	let fsc = CString::new(pathname.as_bytes())?;
+	let mut page: Vec<u8> = vec![0; std::mem::size_of::<u8>() * 16];
```
[the remark this answers is not in the document]
Comment (beevans) [reply liblustreapi/src/lib.rs]: done

File: liblustreapi/src/lib.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (beevans) [reply liblustreapi/src/lib.rs:278]: So open a ticket to add SymbolNotFound to LiblustreError and handle all the cases in here?

[the remark this answers is not in the document]
Comment (jgrund) [reply liblustreapi/src/lib.rs:278]: New ticket and fix in here is fine.

[the remark this answers is not in the document]
Comment (beevans) [reply liblustreapi/src/lib.rs:278]: #2226 created

**Review by utopiabound — DISMISSED** [review]

**Comment by jgrund** [comment]
This needs rebase on latest master due to: https://github.com/whamcloud/integrated-manager-for-lustre/pull/2223

File: liblustreapi/src/lib.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (jgrund) [reply liblustreapi/src/lib.rs:278]: PR: https://github.com/whamcloud/integrated-manager-for-lustre/pull/2228

**Review by jgrund — APPROVED** [review]

**Review by utopiabound — APPROVED** [review]

**Pull request merged**